exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -716,6 +726,7 @@ public String toString() {
     return "ParquetGroupScan [entries=" + entries
         + ", selectionRoot=" + selectionRoot
         + ", numFiles=" + getEntries().size()
+        + ", usedCache=" + usedMetadataCache
```

Comment:
Sounds good.  I'll make sure the docs are consistent with the behavior.
Could you change the name to 'usedMetadataFile' (instead of 'use').   
Rest of 3887 looks good to me.  +1 .    I have not reviewed the other changes.